be/src/exec/vectorized/topn_node.cpp

```diff
@@ -220,6 +217,8 @@ pipeline::OpFactories TopNNode::decompose_to_pipeline(pipeline::PipelineBuilderC
         operators_sink_with_sort.emplace_back(std::move(partition_sort_sink_operator));
         context->add_pipeline(operators_sink_with_sort);
         // Each PartitionSortSinkOperator has an independent LocalMergeSortSinkOperator respectively
+        auto degree_of_parallelism =
+                down_cast<SourceOperatorFactory*>(operators_sink_with_sort[0].get())->degree_of_parallelism();
         local_merge_sort_source_operator->set_degree_of_parallelism(degree_of_parallelism);
```

Comment:
Could we just use the DOP from the source operator of `operators_sink_with_sort` after `maybe_interpolate_local_shuffle_exchange()`, and pass it to `SortContextFactory`? 
In this way , maybe we needn't modify `SortContextFactory`.

Reply:
+1

Reply:
Absolutely